rfq/order.go

```diff
@@ -314,7 +333,7 @@ func (c *AssetPurchasePolicy) CheckHtlcCompliance(
 
 	// Ensure that the outbound HTLC amount is less than the maximum agreed
 	// BTC payment.
-	if htlc.AmountOutMsat > c.PaymentMaxAmt {
+	if (c.CurrentAmountMsat + htlc.AmountOutMsat) > c.PaymentMaxAmt {
```

Comment:
This would work for a single attempt. But in case where multiple attempts over the same edge node are done, this would quickly result in the HTLCs being rejected, because we would never reduce the current amount based on the HTLCs that were canceled from previous attempts.